assets/js/blocks/product-gallery/frontend.tsx

```diff
@@ -1,221 +1,161 @@
 /**
  * External dependencies
  */
-import { store as interactivityApiStore } from '@woocommerce/interactivity';
-
-interface State {
-	[ key: string ]: unknown;
-}
-
-export interface ProductGalleryInteractivityApiContext {
-	woocommerce: {
-		selectedImage: string;
-		imageId: string;
-		visibleImagesIds: string[];
-		dialogVisibleImagesIds: string[];
-		isDialogOpen: boolean;
-		productId: string;
-	};
-}
-
-export interface ProductGallerySelectors {
-	woocommerce: {
-		isSelected: ( store: unknown ) => boolean;
-		pagerDotFillOpacity: ( store: SelectorsStore ) => number;
-		selectedImageIndex: ( store: SelectorsStore ) => number;
-		isDialogOpen: ( store: unknown ) => boolean;
-	};
-}
-
-interface Actions {
-	woocommerce: {
-		thumbnails: {
-			handleClick: (
-				context: ProductGalleryInteractivityApiContext
-			) => void;
-		};
-		handlePreviousImageButtonClick: {
-			( store: Store ): void;
-		};
-		handleNextImageButtonClick: {
-			( store: Store ): void;
-		};
-	};
-}
-
-interface Store {
-	state: State;
-	context: ProductGalleryInteractivityApiContext;
-	selectors: ProductGallerySelectors;
-	actions: Actions;
-	ref?: HTMLElement;
+import { store, getContext as getContextFn } from '@woocommerce/interactivity';
+import { StorePart } from '@woocommerce/utils';
+
+export interface ProductGalleryContext {
+	selectedImage: string;
+	imageId: string;
+	visibleImagesIds: string[];
+	dialogVisibleImagesIds: string[];
+	isDialogOpen: boolean;
+	productId: string;
 }
 
 interface Event {
 	keyCode: number;
 }
```

Comment:
This wouldn't be necessary if the `KeyboardEvent` class is used as the type in the callback below. 👇

In addition, the `keyCode` property is deprecated―it should be replaced with `key` or `code`. See [KeyboardEvent/keyCode](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode).